### ISSUE TITLE
Implement Efficient HTTP Streaming

### ISSUE THREAD

**Issue by frankhinek** [TBD54566975/web5-js#12]
Currently, when data is transported from a client app using the `web5-js` lib to a remote agent via HTTP, the agent temporarily reads the data from a ReadableStream into system memory before persisting to it's embedded DWN.  To support large file transfers (>2GB) in an efficient manner, this approach should be replaced with one that reads that received data in chunks while computing the CID and writing to the embedded DWN.

One challenge that has been encountered is ensuring that the same CID is calculated regardless of whether this is done in browser in Node.js.  The following error is thrown when providing the previously computed CID in a RecordsWrite message that doesn't match what the DWN SDK computes:
```shell
{
  status: {
    code: 400,
    detail: 'MessageStoreDataCidMismatch: actual data CID bafybeif7ztnhq65lumvvtr4ekcwd2ifwgm3awq4zfr3srh462rwyinlb4y does not match dataCid in descriptor: bafybeibri6pn4vauwghjebpqvmb3rt5mal2misnotkmt7si4toemqtx3zq'
  },
  entries: undefined
}
```

**Comment by dcrousso** [comment]
i wrote down an idea for doing this in [dwn-server#8](https://github.com/TBD54566975/dwn-server/issues/8#issuecomment-1475088146) and then gave it a shot in [dwn-sdk-js#280](https://github.com/TBD54566975/dwn-sdk-js/pull/280) :)

**Comment by mistermoe** [comment]
https://github.com/TBD54566975/dwn-chonker-experiment has a working example of producing the same cid in both the browser and node runtimes

**Comment by frankhinek** [comment]
Fixed with [v0.7.0](https://github.com/TBD54566975/web5-js/releases/tag/v0.7.0) release.
